### ISSUE TITLE
Eye kills itself

### ISSUE THREAD

**Issue by hackeron** [kostya/eye#62]
I have this check:

```ruby
require 'rtsp/client'
class Rtsp < Eye::Checker::CustomDefer
  param :addr, String, true

  def initialize(*args)
    super
    @addr = addr
    @rtsp_client = RTSP::Client.new(@addr)
  end

  def get_value
    begin
      if @rtsp_client.describe.code == 200
        check_with_ffmpeg
      else
        false
      end
    rescue
      false
    end
  end

  def good?(value)
    value
  end

  def human_value(value)
    value == true ? 'Ok' : 'Err'
  end

  private
  def check_with_ffmpeg
    system("ffmpeg -loglevel warning -y -stimeout 5000000 " \
      "-i #{@addr} -c:v copy -an -vframes 1 -f rawvideo /dev/null")
  end
end
```

It works most of the time, however sometimes, I get this in eye.log and it stops monitoring all processes:

```
[2014-05-29 21:55:57 +0100] [celluloid] Terminating task: type=:timer, meta=nil, status=:receiving
[2014-05-29 21:55:57 +0100] [recorder:537b68987a616e64da431e00:proxy] check:rtsp task was terminated ["/home/deployer/xanagent/shared/bundle/ruby/2.1.0/gems/celluloid-0.15.2/lib/celluloid/tasks/task_fiber.rb:32:in `terminate'", "/home/deployer/xanagent/shared/bundle/ruby/2.1.0/gems/celluloid-0.15.2/lib/celluloid/actor.rb:404:in `block in cleanup'", "/home/deployer/xanagent/shared/bundle/ruby/2.1.0/gems/celluloid-0.15.2/lib/celluloid/actor.rb:404:in `each'", "/home/deployer/xanagent/shared/bundle/ruby/2.1.0/gems/celluloid-0.15.2/lib/celluloid/actor.rb:404:in `cleanup'", "/home/deployer/xanagent/shared/bundle/ruby/2.1.0/gems/celluloid-0.15.2/lib/celluloid/actor.rb:375:in `shutdown'", "/home/deployer/xanagent/shared/bundle/ruby/2.1.0/gems/celluloid-0.15.2/lib/celluloid/actor.rb:185:in `run'", "/home/deployer/xanagent/shared/bundle/ruby/2.1.0/gems/celluloid-0.15.2/lib/celluloid/actor.rb:157:in `block in initialize'", "/home/deployer/xanagent/shared/bundle/ruby/2.1.0/gems/celluloid-0.15.2/lib/celluloid/thread_handle.rb:13:in `block in initialize'", "/home/deployer/xanagent/shared/bundle/ruby/2.1.0/gems/celluloid-0.15.2/lib/celluloid/internal_pool.rb:100:in `call'", "/home/deployer/xanagent/shared/bundle/ruby/2.1.0/gems/celluloid-0.15.2/lib/celluloid/internal_pool.rb:100:in `block in create'"]
```

The formatted output looks like this:

```
[2014-05-29 21:55:57 +0100] [celluloid] Terminating task: type=:timer, meta=nil, status=:receiving
[2014-05-29 21:55:57 +0100] [recorder:537b68e17a616e64de651e00:proxy] check:rtsp task was terminated
celluloid-0.15.2/lib/celluloid/tasks/task_fiber.rb:32:in `terminate'
celluloid-0.15.2/lib/celluloid/actor.rb:404:in `block in cleanup'
celluloid-0.15.2/lib/celluloid/actor.rb:404:in `each'
celluloid-0.15.2/lib/celluloid/actor.rb:404:in `cleanup'
celluloid-0.15.2/lib/celluloid/actor.rb:375:in `shutdown'
celluloid-0.15.2/lib/celluloid/actor.rb:185:in `run'
celluloid-0.15.2/lib/celluloid/actor.rb:157:in `block in initialize'
celluloid-0.15.2/lib/celluloid/thread_handle.rb:13:in `block in initialize'
celluloid-0.15.2/lib/celluloid/internal_pool.rb:100:in `call'
celluloid-0.15.2/lib/celluloid/internal_pool.rb:100:in `block in create'
```

Any ideas why it happened and how to prevent this from happening?

**Comment by hackeron** [comment]
Hey :) - I had that happen again :(

```
[2015-01-30 14:07:49 +0000] [recorder:52121f8ac5399f1bc100dc10:recorder] process_really_running?: <9615> {:result=>1}
[2015-01-30 14:07:49 +0000] [recorder:52121f8ac5399f1bc100dc10:recorder] process <9615> did not die after TERM, sending KILL
```

Boom, no more eye :( - happened after an unclean reboot, so I'm guessing eye assumed it is still running and was looking for processes based on PID only :(

Any update on this?

**Comment by kostya** [comment]
there is commit for this https://github.com/kostya/eye/commit/ea964b6a7b6027e2e4e272d0133f767f9ae148d7, but it makes eye 2 more times using cpu, need to decrease it.

**Comment by hackeron** [comment]
Thank you for working on this :)

**Comment by kostya** [comment]
this should be fully fixed in 0.8
